### PR TITLE
H-06 Redux advanced homework: Main change

### PR DESCRIPTION
**Pull request by elegorod**
Also fixed bugs and typos. Log server now prints JSON as expected.

### CODE REVIEW

File: homeworks/06-redux-advanced/src/logSender.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  [PizzaActionType.REMOVE_FROM_BASKET, "PIZZA_REMOVED_FROM_BASKET"]
+])
+const UnknownEventName = "UNKNOWN_EVENT"
+
+function createLogEvent(action: PizzaAction, state: State): LogEvent {
+  const eventName = eventNames.get(action.type) ?? UnknownEventName
+  const pizzaId = action.pizzaId
+  const pizza = pizzaId ? state.pizza.find(_ => _._id === pizzaId) : undefined
+  let event
+  if (pizza) {
```
Comment (vladislavkovaliov) [inline homeworks/06-redux-advanced/src/logSender.ts:32]: there can be just.
```
const pizza = state.find(p => p.id === pizzaId);
const event = {
  eventName: eventName,
};
if (pizza) {
  event.pizzaName = pizza.name;
  event.pizzaPrice = pizza.price;
}
```
`event` is object. so, we are able to add new fields.

File: homeworks/06-redux-advanced/src/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+export interface PizzaAction extends Action<PizzaActionType> {
+  pizzaId?: string,
+  pizzas?: Pizza[]
+}
+
+const pizzaComparator = (a: Pizza, b: Pizza) => a.name.localeCompare(b.name)
+
+export const reducer: Reducer<State, PizzaAction> = (state = { pizza: [], basket: [] }, action) => {
+  let newState = state
+  if (action.type === PizzaActionType.LOAD_LIST) {
```
Comment (vladislavkovaliov) [inline homeworks/06-redux-advanced/src/reducer.ts]: Would it better just?
```
if (action.type === PizzaActionType.LOAD_LIST && action.pizzas) {}
```
anyway i would recommend to use switch for action type within reducer as well due to having lots if/else

File: homeworks/06-redux-advanced/src/logSender.ts (hunk 1 above)
Comment (elegorod) [reply homeworks/06-redux-advanced/src/logSender.ts:32]: It doesn't work, show an error

> Property 'pizzaName' does not exist on type '{ eventName: string; }'.ts(2339)

File: homeworks/06-redux-advanced/src/reducer.ts (hunk 2 above)
Comment (elegorod) [reply homeworks/06-redux-advanced/src/reducer.ts]: Switch looks bad when cases are long. So, when chaging to switch, I extracted add and remove handling to separate functions.

**Pull request merged**